src/main/java/pl/edu/agh/student/controller/EventController.java

```diff
@@ -32,8 +30,21 @@ public EventDto save(@RequestBody @Valid EventDto event, BindingResult bindingRe
 
     @RequestMapping(method = RequestMethod.GET)
     @ResponseBody
-    public List<EventDto> getAll() {
+    public List<EventDto> getAllLocal() {
         return eventService.getAll();
     }
 
+    @RequestMapping(value = "/remote", method = RequestMethod.GET)
+    @ResponseBody
+    public List<EventDto> getNotSynchronized(HttpServletRequest request) {
+        return eventService.getNotSynchronized(request);
```

Comment:
Events are duplicate (probably by this and getAllLocal methods) when any of property changed (color for example).